pkg/sink/kafka/sarama.go

```diff
@@ -122,6 +117,28 @@ func NewSaramaConfig(ctx context.Context, o *Options) (*sarama.Config, error) {
 		return nil, cerror.WrapError(cerror.ErrKafkaInvalidConfig, err)
 	}
 
+	kafkaVersion, err := GetKafkaVersion(ctx, config, o)
```

Comment:
move to line 131,   only detect the kafka cluster version when there is no version specified  by the user